zap-widget/src/components/SimpleBoostWrapper.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useRef } from 'react'
+import 'simple-boost'
+import { useClient } from '../context'
+
+const SimpleBoostWrapper = ({
+  address,
+  amount = 1,
+  currency = 'usd',
+  className = '',
+}: {
+  address: string
+  amount?: number
+  currency?: string
+  className?: string
+}) => {
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const { setInvoice } = useClient()
+  useEffect(() => {
+    let boostEl: Element | null = null
+    const handleSuccess = (e: Event) => {
+      const customEvent = e as CustomEvent<{ pr: string, preimage: string }>
+      setInvoice(customEvent.detail);
```

Comment:
You don't use the preimage anywhere, so you can set Invoice to string type instead of `{ pr: string, preimage: string }`